src/net_processing.cpp

```diff
@@ -949,6 +949,16 @@ void static ProcessGetData(CNode* pfrom, const Consensus::Params& consensusParam
                 BlockMap::iterator mi = mapBlockIndex.find(inv.hash);
                 if (mi != mapBlockIndex.end())
                 {
+                    if (mi->second->nChainTx && !mi->second->IsValid(BLOCK_VALID_SCRIPTS) &&
```

Comment:
I should think this through a bit more, but I believe the requirement that the block be VALID_SCRIPTS is stronger than is needed -- in particular, if two blocks arrive simultaneously, we might relay both before picking which of the two is our tip, and then gating ProcessGetData() on VALID_SCRIPTS will cause us to stall a peer who picked the other.

Perhaps we should change VALID_SCRIPTS to VALID_TRANSACTIONS below, and then actually I'm not sure if we would still need this code path?